### PR TITLE
Upgrade Django to 4.2

### PR DESCRIPTION
**Pull request by Dhwaniartefact**
This sets django-upgrade in pre-commit to upgrade Django to 4.2.

One significant change is: 
The Django 4.2 version uses a datetime method called fromisoformat() to get well-formed values for timezone in parse_datetime() of dateparse module.
Ref: https://github.com/django/django/blob/main/django/utils/dateparse.py#L104

Connected to https://github.com/archivematica/Issues/issues/1624

### CODE REVIEW

File: src/dashboard/src/components/administration/tests/test_dip_upload_configs.py [hunk 1]
```diff
@@ -60,15 +60,23 @@ def test_post_missing_fields(self):
         self.assertFalse(form.is_valid())
         self.assertTrue(form.errors)
 
-        self.assertFormError(response, "form", "user", "This field is required.")
-        self.assertFormError(response, "form", "xlink_show", "This field is required.")
         self.assertFormError(
-            response, "form", "xlink_actuate", "This field is required."
+            response.context["form"], "user", "This field is required."
```
Comment (replaceafill) [inline src/dashboard/src/components/administration/tests/test_dip_upload_configs.py:64]: This change is documented in https://docs.djangoproject.com/en/4.2/releases/4.1/#id2 under:
>Passing a response object and a form/formset name to SimpleTestCase.assertFormError() and assertFormsetError() is deprecated

File: src/dashboard/src/settings/base.py [hunk 2]
```diff
@@ -292,11 +292,9 @@ def _get_settings_from_file(path):
 
 # If you set this to False, Django will make some optimizations so as not
 # to load the internationalization machinery.
```
Comment (replaceafill) [inline src/dashboard/src/settings/base.py]: Can these be removed?

File: src/dashboard/src/settings/base.py [hunk 3]
```diff
@@ -292,11 +292,9 @@ def _get_settings_from_file(path):
...
-USE_I18N = True
 
 # If you set this to False, Django will not format dates, numbers and
 # calendars according to the current locale
```
Comment (replaceafill) [inline src/dashboard/src/settings/base.py]: Can these be removed?

File: src/dashboard/src/components/administration/tests/test_dip_upload_configs.py (hunk 1 above)
Comment (Dhwaniartefact) [reply src/dashboard/src/components/administration/tests/test_dip_upload_configs.py:64]: Got it. Thank you !

**Comment by replaceafill** [comment]
This looks good @Dhwaniartefact, thanks! Let's leave this PR here until we milestone its issue.

**Review by replaceafill — APPROVED** [review]
Great work @Dhwaniartefact!

**Pull request merged**